### ISSUE TITLE
Prefetch icons for other states

### ISSUE THREAD

**Issue by mueller-ma** [openhab/openhab-android#684]
Currently the icon for a specific state is fetched, when the state appears the first time. Maybe we should load icons for other states on the first app start, e.g. on and off for switches, 0 to 100 for slider.
This would avoid flickering icons, not related to #669

**Comment by maniac103** [comment]
I'm not sure whether that really is a good idea, because of the amount of data that potentially is transferred (think 100 slider icons) which then are somewhat likely to never be shown...

**Comment by mueller-ma** [comment]
True. Another fix could be to show the old icon while the new loads. This will at least fix the flicker. Might be good for charts as well

**Comment by maniac103** [comment]
That indeed sounds like a good idea. There's two types of replacement icons in the SmartImageView code: fallback and progress. We should make it display
- progress while loading if defined
- fallback if no drawable is set currently or if loading a new drawable failed
- previous image otherwise

**Comment by PartTimeDev** [comment]
I have the flickering even for single state icons.
Socket icon in this case

![screencast_2018-05-30-19-59-29](https://user-images.githubusercontent.com/39832934/40844991-a9d32440-65b5-11e8-820c-4ab991fd091d.gif)

**Comment by mueller-ma** [comment]
A new icon will be loaded for every new state, e.g. 272, 255, 253 even when the icon is the same. If they are loaded once, they won't flicker again. Look at "Hausverbrauch" when it changes to 253W. No flicker, because the 253-icon is already cached.

**Comment by maniac103** [comment]
@mueller-ma Maybe opening a feature request towards the server to provide information whether the given icon depends on state or not would be an idea or not? The current state of things is a bit inefficient in that regard.

**Comment by mueller-ma** [comment]
That's a good idea since there is so much overhead when you look at items with a large range of values.
